deepchem/models/torch_models/layers.py

```diff
@@ -4792,3 +4792,276 @@ def forward(self, dist: torch.Tensor) -> torch.Tensor:
+    def forward(self, node_features: torch.Tensor, rbf: torch.Tensor,
+                sbf1: torch.Tensor, sbf2: torch.Tensor, idx_kj: torch.Tensor,
+                idx_ji_1: torch.Tensor, idx_jj: torch.Tensor,
+                idx_ji_2: torch.Tensor,
+                edge_index: torch.Tensor) -> Tuple[torch.Tensor, torch.Tensor]:
+        """The forward method performs the computation for the MXMNetLocalMessagePassing Layer.
+
+        This method processes the input tensors representing atom features, radial basis functions (RBF), and spherical basis functions (SBF) using message passing over the molecular graph. The message passing updates the atom representations, and the resulting tensor represents the updated atom feature after local message passing.
+
+        Parameters
+        ----------
+        node_features : torch.Tensor
+            Input tensor representing atom features.
+        rbf : torch.Tensor
+            Input tensor representing radial basis functions.
+        sbf1 : torch.Tensor
+            Input tensor representing the first set of spherical basis functions.
+        sbf2 : torch.Tensor
+            Input tensor representing the second set of spherical basis functions.
+        idx_kj : torch.Tensor
+            Tensor containing indices for the k and j atoms involved in each interaction.
+        idx_ji_1 : torch.Tensor
+            Tensor containing indices for the j and i atoms involved in the first message passing step.
+        idx_jj : torch.Tensor
+            Tensor containing indices for the j and j' atoms involved in the second message passing step.
+        idx_ji_2 : torch.Tensor
+            Tensor containing indices for the j and i atoms involved in the second message passing step.
+        edge_index : torch.Tensor
+            Tensor containing the edge indices of the molecular graph, with shape (2, M), where M is the number of edges.
+
+        Returns
+        -------
+        node_features, output: Tuple[torch.Tensor, torch.Tensor]
+            A tuple containing two output tensors:
+            - Updated atom representations after local message passing.
+            - Output tensor representing a fixed-size representation, with shape (N, 1).
+        """
+
+        residual_node_features: torch.Tensor = node_features
+
+        # Integrate the Cross Layer Mapping inside the Local Message Passing
+        node_features = self.h_mlp(node_features)
+
+        # Message Passing 1
+        j, i = edge_index
+        m: torch.Tensor = torch.cat([node_features[i], node_features[j], rbf],
+                                    dim=-1)
+
+        m_kj: torch.Tensor = self.mlp_kj(m)
+        m_kj = m_kj * self.lin_rbf1(rbf)
+        m_kj = m_kj[idx_kj] * self.mlp_sbf1(sbf1)
+        m_kj = scatter(m_kj, idx_ji_1, dim=0, dim_size=m.size(0), reduce='add')
+
+        m_ji_1: torch.Tensor = self.mlp_ji_1(m)
+
+        m = m_ji_1 + m_kj
+
+        # Message Passing 2       (index jj denotes j'i in the main paper)
```

Comment:
please check the comment here. `j'i` what is the meaning of this notation?

Reply:
I think it means message passing between atoms `j'` and `i` based on `forward` docstring

Reply:
Please improve this comment for a better interpretation

Reply:
j' represents other neighbour nodes of i to get the other local angles.